docs/porting-guide.rst

```diff
@@ -473,6 +473,36 @@ constants must also be defined:
    Defines the total size of the physical address space in bytes. For example,
    for a 32 bit physical address space, this value should be ``(1ull << 32)``.
 
```

Comment:
This patch needs further discussion and should in any case be separate from the rest of this PR.

If memory footprint is the concern, then it should be possible for the platform to overlay BL32 RW data over BL2 so the MMU tables for those 2 images at least don't occupy in-use memory at the same time.

I can see your solution would share BL1 xlat table memory too but is there really much overall memory to be saved? The overall PLAT_BASE_XLAT_SIZE has to be large enough for the largest BL's needs. I expect this to be BL2, which is not resident at runtime. So the delta in the number of translation tables needed between BL2 and BL32 is wasted.